### ISSUE TITLE
gf_m2ts_process_packet: Assertion `af_size<=182' failed.

### ISSUE THREAD

**Issue by jocooza** [gpac/gpac#211]
I have built MP4Box on Ubuntu 12.04 64-bits following these steps:

https://gpac.wp.mines-telecom.fr/2015/07/29/gpac-build-mp4box-only-all-platforms/

I then try to repackage a TS file into a MP4 file:

    $ MP4Box -add in.ts out.mp4

    [MPEG-2 TS] Estimated duration based on instant bitrate: 11811.6 sec
    Importing MPEG-2 TS (PID 256)
    MPEG-4 AVC/H264 Video import (TS PID 256)
    [MPEG-2 TS] Estimated duration based on instant bitrate: 11811.6 sec
    [MPEG-2 TS] Estimated duration based on instant bitrate: 14370 sec
    [MPEG-2 TS] Estimated duration based on instant bitrate: 16180.9 sec
    [MPEG-2 TS] Estimated duration based on instant bitrate: 15020.8 sec
    MP4Box: media_tools/mpegts.c:3212: gf_m2ts_process_packet: Assertion `af_size<=182' failed.
    Aborted

**Comment by rbouqueau** [comment]
Thanks for reporting. Would it be possible to share the file?

**Comment by jocooza** [comment]
http://uptobox.com/n1hk6fmx3js5

**Comment by rbouqueau** [comment]
Thank you. The file is not compliant with this rule (ITU-T.Rec.H.222.0 (2006 E)):
```When the adaptation_field_control value is '11', the value of the adaptation_field_length shall
be in the range 0 to 182```
(int the stream adaptation_field_length is 183)

Do you know how the stream was muxed? I'd like to propose the most appropriate solution depending on the source.

Note to self: also I think GPAC non-debug should compile with the NDEBUG flag to not stop on asserts.

**Comment by rbouqueau** [comment]
Otherwise the execution is all fine regarding that the file is incorrect. Log level is 'warning', which is correct too.

> Note to self: also I think GPAC non-debug should compile with the NDEBUG flag to not stop on asserts.

Done in 71748d72e907ed743264aaf09b398eb9098a1289 . Now it should run until the end of the file (but expect a log of log messages).

@jocooza Does it answer your issue?

**Comment by rbouqueau** [comment]
I did not mention, but the file is correctly imported.

**Comment by jocooza** [comment]
I don't know how the stream was muxed as it is obtained from the national public service broadcaster in my country. All other streams I have tried also have this problem, so their video processing procedure must be flawed.

MP4Box from the Ubuntu 12.04 repository (GPAC version 0.4.6-DEV-rev) do repackage the file successfully, but the output has many artifacts and the messages "reference picture missing during reorder" and "Missing reference picture" are shown multiple times in the video player log.

I should mention that the file I uploade was created by joining 245 individual TS files, but I don't think it makes any difference.

I have built the latest revison and can confirm the file is now being repackaged successfully without any artifacts. As you mentioned, MP4Box generates a large amount of log messages while processing the file, but as long as the output is correct I don't see this as a problem.

Thank you.
